src/UpgradeRule/PHP/ApiChangeWarningsRule.php

```diff
@@ -148,7 +160,11 @@ protected function buildRuleVisitors(MutableSource $source, ItemInterface $file)
             $visitors[] = new ConstantWarningsVisitor($warnings['constants'], $source, $file);
         }
         if (isset($warnings['props'])) {
-            $visitors[] = new PropertyWarningsVisitor($warnings['props'], $source, $file);
+            $visitors[] = new PropertyWarningsVisitor($warnings['props'], $source, $file, $this->options);
+        }
+
+        if ((!isset($this->options['skip-visibility']) || !$this->options['skip-visibility'])) {
```

Comment:
Personally, I would rather keep console interaction within the console command. Basically, `ApiChangeWarningRule` shouldn't need to be aware that it's being called from the console.